services/director/src/simcore_service_director/registry_proxy.py

```diff
@@ -375,48 +430,36 @@ async def get_image_details(
 
 
 async def get_repo_details(app: FastAPI, image_key: str) -> list[dict[str, Any]]:
-
-    image_tags = await list_image_tags(app, image_key)
-
-    results = await limited_gather(
-        *[get_image_details(app, image_key, tag) for tag in image_tags],
-        reraise=False,
-        log=logger,
-        limit=get_application_settings(
-            app
-        ).DIRECTOR_REGISTRY_CLIENT_MAX_CONCURRENT_CALLS,
-    )
-    return [result for result in results if not isinstance(result, BaseException)]
+    repo_details = []
+    async for image_tags in list_image_tags_gen(app, image_key):
+        async for image_details_future in limited_as_completed(
+            (get_image_details(app, image_key, tag) for tag in image_tags),
+            limit=get_application_settings(
+                app
+            ).DIRECTOR_REGISTRY_CLIENT_MAX_CONCURRENT_CALLS,
+        ):
+            with log_catch(_logger, reraise=False):
+                if image_details := await image_details_future:
+                    repo_details.append(image_details)
+    return repo_details
 
 
 async def list_services(app: FastAPI, service_type: ServiceType) -> list[dict]:
-    logger.debug("getting list of services")
-    repos = await _list_repositories(app)
-    # get the services repos
-    prefixes = []
-    if service_type in [ServiceType.DYNAMIC, ServiceType.ALL]:
-        prefixes.append(_get_prefix(ServiceType.DYNAMIC))
-    if service_type in [ServiceType.COMPUTATIONAL, ServiceType.ALL]:
-        prefixes.append(_get_prefix(ServiceType.COMPUTATIONAL))
-    repos = [x for x in repos if str(x).startswith(tuple(prefixes))]
-    logger.debug("retrieved list of repos : %s", repos)
-
-    # only list as service if it actually contains the necessary labels
-    results = await limited_gather(
-        *[get_repo_details(app, repo) for repo in repos],
-        reraise=False,
-        log=logger,
-        limit=get_application_settings(
-            app
-        ).DIRECTOR_REGISTRY_CLIENT_MAX_CONCURRENT_CALLS,
-    )
-
-    return [
-        service
-        for repo_details in results
-        if isinstance(repo_details, list)
-        for service in repo_details
-    ]
+    with log_context(_logger, logging.DEBUG, msg="listing services"):
+        services = []
+        async for repos in _list_repositories_gen(app, service_type):
+            # only list as service if it actually contains the necessary labels
+            async for repo_details_future in limited_as_completed(
+                (get_repo_details(app, repo) for repo in repos),
+                limit=get_application_settings(
```

Comment:
MINOR:  you save two calls per iteration by just putting the constants at the top
```python
limit=get_application_settings( app).DIRECTOR_REGISTRY_CLIENT_MAX_CONCURRENT_CALLS

async for repos in _list(...):
      async for repo_details_future in limited_as_completed( ... , limit=limit)
           ...
```

Reply:
true